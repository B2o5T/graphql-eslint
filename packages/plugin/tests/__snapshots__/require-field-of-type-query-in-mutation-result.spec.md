// Vitest Snapshot v1

exports[`Invalid #2 1`] = `
#### ⌨️ Code

      1 |         type Query
      2 |         type Mutation
      3 |
      4 |         extend type Mutation {
      5 |           createUser: User!
      6 |         }

#### ❌ Error

      4 |         extend type Mutation {
    > 5 |           createUser: User!
        |                       ^^^^ Mutation result type "User" must contain field of type "Query"
      6 |         }
`;

exports[`Invalid #3 1`] = `
#### ⌨️ Code

      1 |         type RootQuery
      2 |         type RootMutation {
      3 |           createUser: [User]
      4 |         }
      5 |
      6 |         schema {
      7 |           mutation: RootMutation
      8 |           query: RootQuery
      9 |         }

#### ❌ Error

      2 |         type RootMutation {
    > 3 |           createUser: [User]
        |                        ^^^^ Mutation result type "User" must contain field of type "RootQuery"
      4 |         }
`;

exports[`Invalid #4 1`] = `
#### ⌨️ Code

       1 |         type RootQuery
       2 |         type RootMutation
       3 |         extend type RootMutation {
       4 |           createUser: [User!]!
       5 |         }
       6 |
       7 |         schema {
       8 |           mutation: RootMutation
       9 |           query: RootQuery
      10 |         }

#### ❌ Error

      3 |         extend type RootMutation {
    > 4 |           createUser: [User!]!
        |                        ^^^^ Mutation result type "User" must contain field of type "RootQuery"
      5 |         }
`;

exports[`should ignore arguments 1`] = `
#### ⌨️ Code

      1 |         type Query
      2 |         type Mutation {
      3 |           createUser(a: User, b: User!, c: [User], d: [User]!, e: [User!]!): User
      4 |         }

#### ❌ Error

      2 |         type Mutation {
    > 3 |           createUser(a: User, b: User!, c: [User], d: [User]!, e: [User!]!): User
        |                                                                              ^^^^ Mutation result type "User" must contain field of type "Query"
      4 |         }
`;
