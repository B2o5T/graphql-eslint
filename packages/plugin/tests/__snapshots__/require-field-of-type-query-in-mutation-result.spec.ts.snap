// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
  1 |
  2 |         type Query
  3 |         type Mutation {
> 4 |           createUser: User!
    |                       ^^^^ Mutation result type "User" must contain field of type "Query".
  5 |         }
  6 |       
`;

exports[` 2`] = `
  1 |
  2 |         type Query
  3 |         type Mutation
  4 |
  5 |         extend type Mutation {
> 6 |           createUser: User!
    |                       ^^^^ Mutation result type "User" must contain field of type "Query".
  7 |         }
  8 |       
`;

exports[` 3`] = `
   1 |
   2 |         type RootQuery
   3 |         type RootMutation {
>  4 |           createUser: User!
     |                       ^^^^ Mutation result type "User" must contain field of type "RootQuery".
   5 |         }
   6 |
   7 |         schema {
   8 |           mutation: RootMutation
   9 |           query: RootQuery
  10 |         }
  11 |       
`;

exports[` 4`] = `
   1 |
   2 |         type RootQuery
   3 |         type RootMutation
   4 |         extend type RootMutation {
>  5 |           createUser: User!
     |                       ^^^^ Mutation result type "User" must contain field of type "RootQuery".
   6 |         }
   7 |
   8 |         schema {
   9 |           mutation: RootMutation
  10 |           query: RootQuery
  11 |         }
  12 |       
`;
